plugin-notes/notes.cpp

```diff
@@ -0,0 +1,252 @@
+/* BEGIN_COMMON_COPYRIGHT_HEADER
+ * (c)LGPL2+
+ *
+ * LXQt - a lightweight, Qt based, desktop toolset
+ * https://lxqt.org
+ *
+ * Copyright: 2012 Razor team
+ * Authors:
+ *   Aaron Lewis <the.warl0ck.1989@gmail.com>
+ *
+ * This program or library is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General
+ * Public License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA
+ *
+ * END_COMMON_COPYRIGHT_HEADER */
+
+#include "notes.h"
+#include "notesconfiguration.h"
+#include "stickynote.h"
+
+#include <QMouseEvent>
+#include <QResizeEvent>
+#include <QHBoxLayout>
+#include <QVBoxLayout>
+#include <QScreen>
+#include <QStandardPaths>
+#include <QSettings>
+#include <QLabel>
+#include <QDateTime>
+#include <QMessageBox>
+#include <QDir>
+#include <QFontDialog>
+#include <QAction>
+#include <QMenu>
+#include <QDebug>
+
+Notes::Notes(const ILXQtPanelPluginStartupInfo &startupInfo) :
+    QObject(),
+    ILXQtPanelPlugin(startupInfo),
+    mHidden(false)
+{
+    realign();
+    
+    mButton.setAutoRaise(true);
+    mButton.setIcon(XdgIcon::fromTheme("date", "date"));
+    
+    QAction *addNew = new QAction(tr("New Note"));
+    QAction *showHide = new QAction(tr("Show/Hide Notes"));
+    connect(addNew, SIGNAL(triggered()), this, SLOT(addNewNote()));
+    connect(showHide, SIGNAL(triggered()), this, SLOT(toggleShowHide()));
+    
+    QMenu *menu = new QMenu;
+    menu->addAction(addNew);
+    menu->addAction(showHide);
+    mButton.setMenu(menu);
+    mButton.setPopupMode(QToolButton::InstantPopup);
+    
+    QString showOnStartup = settings()->value("showNotesOnStartup","").toString();
```

Comment:
Please, use default values as in others places (this `toString()` `if` `toBool()` equilibristics is useless IMO.